commands/run.go

```diff
@@ -39,26 +48,48 @@ var RunCmd = cli.Command{
 			return err
 		}
 
-		nodeRange, args := parseCommand(c.Args(), c.IsSet("terminator"))
-
-		if nodeRange == "" {
-			nodeRange = fmt.Sprintf("[0-%d]", len(nodes)-1)
-		}
-
-		list, err := parseRange(nodeRange)
-		if err != nil {
-			return fmt.Errorf("could not parse node range %s", nodeRange)
+		var args [][]string
+		var terminatorPresent []bool
+		if c.IsSet("stdin") {
+			scanner := bufio.NewScanner(os.Stdin)
+			for scanner.Scan() {
+				tokens := strings.Fields(scanner.Text())
+				term := tokens[0] == "--"
+				if term {
+					tokens = tokens[1:]
+				}
+				terminatorPresent = append(terminatorPresent, term)
+				args = append(args, tokens)
+			}
+		} else {
+			cArgsStr := make([]string, c.NArg())
+			for i, arg := range c.Args() {
+				cArgsStr[i] = arg
+			}
+			args = append(args, cArgsStr)
+			terminatorPresent = append(terminatorPresent, c.IsSet("terminator"))
```

Comment:
Something I think we can do here to make this simpler is to normalize everything to an `io.Reader` that we can pass to `bufio.NewScanner` regardless of using stdin, or passing through arguments.

So for the non-stdin case, we construct an `io.Reader` using [`strings.NewReader`](https://golang.org/pkg/strings/#NewReader), we can then just hand it to `bufio.NewScanner` and let it be handled the exact same way, allowing us to move that scanner logic out, and into the main execution path. This also helps clean up the some of the terminator logic, as we can just assume everything will have it present (we can inject it back in on the args case).

We only need the `terminator` flag for the cli args case, as the cli library will swallow it, in the stdin case, the flag terminator is always present (if the user supplied it, if they didn't it's an error anyways).

Reply:
Hm, I like the abstraction but to make it happen I think we would have to reconstruct the input argument into a single string in order to pass it to `strings.NewReader` in the non-stdin case. This means that `cli` would break up the string for us, we'd put it back together, then take it apart again once we processes the command. Seems a bit roundabout, but I'm not sure how else to make that happen since we don't seem to have access to the original string once we're in the `Action` function.

Reply:
Also, is it necessarily the case that not passing `--` should result in an error? Seems like the following could be supported:

```
iptb run <<CMD
0 -- "Running on node 0"
echo "Running on all nodes"
CMD
```

Depends on what we want here, though.

Reply:
I think the abstraction is worth it in this case, working on it now.

Reply:
And re:

>It would be nice to make any line of the cmd file be valid arguments to the command itself.

Agreed with that, I was going for the same but was forgetting how `iptb run` worked.